lib/ansible/context.py

```diff
@@ -0,0 +1,53 @@
+# Copyright: (c) 2018, Toshio Kuratomi <tkuratomi@ansible.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+"""
+Context of the running Ansible.
+
+In the future we *may* create Context objects to allow running multiple Ansible plays in parallel
+with different contexts but that is currently out of scope as the Ansible library is just for
+running the ansible command line tools.
+
+These APIs are still in flux so do not use them unless you are willing to update them with every Ansible release
+"""
+
+from ansible.arguments.context_objects import CLIArgs, GlobalCLIArgs
+
+
+# Note: this is not the singleton version.  That is only created once the program has actually
+# parsed the args
+CLIARGS = CLIArgs({})
+
+
+class _Context:
+    """
+    Not yet ready for Prime Time
+
+    Eventually this may allow for code which needs to run under different contexts (for instance, as
+    if they were run with different command line args or from different current working directories)
+    to exist in the same process.  But at the moment, we don't need that so this code has not been
+    tested for suitability.
+    """
+    def __init__(self):
+        global CLIARGS
```

Comment:
Do we need this `global` here?

Reply:
We don't.  I think there was a linter or a recommendation from someone to always use global when referencing a global that can be changed even if you aren't doing it in that function.  This is probably on the edge of even that recommendation, though, since it's both a Singleton and only the _init_global_context() function should be changing it.

I'll get rid of this.

Reply:
Done in latest version